cmd/snap/cmd_info.go

```diff
@@ -321,21 +321,28 @@ func maybePrintServices(w io.Writer, snapName string, allApps []client.AppInfo,
 
 var channelRisks = []string{"stable", "candidate", "beta", "edge"}
 
-// displayChannels displays channels and tracks in the right order
-func (x *infoCmd) displayChannels(w io.Writer, chantpl string, esc *escapes, remote *client.Snap, revLen, sizeLen int) (maxRevLen, maxSizeLen int) {
-	fmt.Fprintln(w, "channels:")
+type channelInfo struct {
+	indent, name, version, released, revision, size, notes string
+}
 
-	releasedfmt := "2006-01-02"
-	if x.AbsTime {
-		releasedfmt = time.RFC3339
-	}
+type channelInfos struct {
+	channels              []*channelInfo
+	maxRevLen, maxSizeLen int
+	releasedfmt, chantpl  string
+	needsHeader           bool
+}
 
-	type chInfoT struct {
-		name, version, released, revision, size, notes string
+func (chInfos *channelInfos) addOne(chInfo *channelInfo) {
```

Comment:
feels like this should make the channelInfo and not take it,  because otherwise we still have the SizeToStr and some formatting outside of this.

Maybe it's overkill but I would,   set esc on the struct,

* add() taking arguments and doing a bit more formatting and creating the channelInfo

* addChannelSnapInfo
* addEmptyChannel
* addInstalled

built on top

Reply:
I think I got what you meant, please confirm :-)